core/trino-main/src/test/java/io/trino/operator/scalar/FunctionAssertions.java

```diff
@@ -307,17 +308,11 @@ private Object selectUniqueValue(String projection, Type expectedType, Session s
         return Iterables.getOnlyElement(resultSet);
     }
 
-    // this is not safe as it catches all RuntimeExceptions
-    @Deprecated
-    public void assertInvalidFunction(String projection)
+    public void assertInvalidFunction(String projection, Class<?> type, String regex)
```

Comment:
I am not convinced about this API

- it's usually more convenient to work with constant `String message` than a regex
- catching some `type` should not be needed -- TrinoException should be typically be expected

Reply:
We should remove this deprecated method since there is only one caller.